### PR TITLE
share link issue for date  enforcement

### PR DESCRIPTION
**Pull request by khushbu-maurya**

### CODE REVIEW

**Comment by admin-vizcentric** [comment]
There were no errors, go have a cup of coffee...

File: src/main/webapp/app/entities/flair-bi/flair-bi.state.js [hunk 1]
```diff
@@ -119,6 +119,22 @@
                             }
                         }
                     ],
+                    entity: [
+                        "$stateParams",
+                        "Views",
+                        "$q",
+                        function($stateParams, Views, $q) {
+                            if (!isNaN($stateParams.viewId)) {
```
Comment (johibkhan2) [inline src/main/webapp/app/entities/flair-bi/flair-bi.state.js:127]: IsNan is not required. Why its included?

File: src/main/webapp/app/entities/flair-bi/flair-bi.state.js [hunk 2]
```diff
@@ -119,6 +119,22 @@
...
+                                return Views.get({
+                                    id: $stateParams.viewId
+                                }).$promise;
+                            } else {
+                                var deferred = $q.defer();
+                                deferred.reject("Not valid id");
```
Comment (johibkhan2) [inline src/main/webapp/app/entities/flair-bi/flair-bi.state.js:133]: Rename the message "it's not valid view id 1235"

File: src/main/webapp/app/entities/flair-bi/fullscreen/flair-bi-fullscreen.controller.js [hunk 3]
```diff
@@ -139,6 +157,113 @@
...
+            var dynamics;
+            var tooltipText;
+            var dynamicDateRangeToolTip;
+            var dynamicDateRangeObject;
+            var daterange = value.split("||");
+            var selected;
+            if (metadata) {
+                dynamics = metadata.split("||");
+                tooltipText = dynamics[0] === "true" ? 'Last ' + dynamics[1] : dynamics[2];
+                dynamicDateRangeToolTip = { name: filterName, text: tooltipText };
```
Comment (johibkhan2) [inline src/main/webapp/app/entities/flair-bi/fullscreen/flair-bi-fullscreen.controller.js:233]: dynamics[0]? Is enough.. No need write ==true

File: src/main/webapp/app/entities/flair-bi/fullscreen/flair-bi-fullscreen.controller.js [hunk 4]
```diff
@@ -139,6 +157,113 @@
...
+                    delete filterParameters[item.name]
+                    addDefaultDateRangeFilter(filterParameters, item.selected, item);
+                    addDefaultDateRangeFilter(filterParameters, item.selected2, item);
+                });
+                filterParametersService.saveSelectedFilter(filterParameters);
+                filterParametersService.save(filterParametersService.getSelectedFilter());
+            }
+        }
+
+        function applyViewFeatureCriteria() {
```
Comment (johibkhan2) [inline src/main/webapp/app/entities/flair-bi/fullscreen/flair-bi-fullscreen.controller.js:195]: This is duplicate function exist in header..check if u can create a common service in dispatcher js file

**Comment by admin-vizcentric** [comment]
There were no errors, go have a cup of coffee...

**Comment by admin-vizcentric** [comment]
There were no errors, go have a cup of coffee...

**Pull request merged**